### ISSUE TITLE
[water] mapnik dosn't use width tag for waterway=canal

### ISSUE THREAD

**Issue by tomhughes** [tomhughes/trac-tickets#1721]
**[Submitted to the original trac issue database at 8.31am, Sunday, 19th April 2009]**

For example:
http://openstreetmap.org/?lat=50.6042&lon=9.46831&zoom=17&layers=B000FTF

**Comment by tomhughes** [comment]
**Author: bmog**
**[Added to the original trac issue at 8.53pm, Sunday, 13th February 2011]**

Replying to [ticket:1721 bmog]:
> For example:
> http://openstreetmap.org/?lat=50.6042&lon=9.46831&zoom=17&layers=B000FTF

Hi,

this bug is still present and very annoying.

Was this bug simply forgotten or is there a more fundamental problem ?
width tag on an artifical object like canal provides practical information, I think !?

**Comment by tomhughes** [comment]
**Author: Ldp**
**[Added to the original trac issue at 1.06pm, Thursday, 15th September 2011]**

The width tag is visually hard to implement. What should render at the transition from sections with width tag and those without, or with different values?

All that could conceivably be done now is to make a sudden change in the width of the rendered line, which is aesthetically displeasing.

A major technical hurdle is the fact that the source data can be in different geographical projections, some in projected values, some in latlong values. Line widths are done in pixels. Somehow the geographical value would need to be translated to a pixel value.

In short, I don't see support for width=* tags in the foreseeable future.

**Comment by tomhughes** [comment]
**Author: cmuelle8**
**[Added to the original trac issue at 4.56pm, Monday, 11th February 2013]**

Replying to [comment:3 Ldp]:
> The width tag is visually hard to implement. What should render at the transition from sections with width tag and those without, or with different values?
> 
> All that could conceivably be done now is to make a sudden change in the width of the rendered line, which is aesthetically displeasing.
> 
> A major technical hurdle is the fact that the source data can be in different geographical projections, some in projected values, some in latlong values. Line widths are done in pixels. Somehow the geographical value would need to be translated to a pixel value.
> 
> In short, I don't see support for width=* tags in the foreseeable future.


Why not just adjust the rendering in ranges, this way you don't have to translate width to an exact pixel value?  E.g.


```
waterway=canal  &&  width less than 2.5m -> render like waterway=ditch
waterway=canal  &&  width less than 7.5m -> render like waterway=stream
waterway=canal  &&  width above 7.5m -> render like waterway=river

of course these values might be consented upon
```

If this is not implemented people "tag for renderers", e.g. tag small man_made canals as waterway=stream, instead of doing it right.  It's understandable since a small canal carrying wood in the medieval ages should not appear visually the same as a large canal built for barges or similar.

Thanks a ton to whomever steps up and fixes this bug in the mapnik stylesheet.



This would most probably also fix #3590 and #4656

**Comment by tomhughes** [comment]
**Author: math1985**
**[Added to the original trac issue at 10.54pm, Monday, 26th May 2014]**

This is difficult to accomplish. For example, what should we do where the width changes? Explicitly mapping the outline would be the best solution. I will close this as wontfix.
